src/axislike.rs

```diff
@@ -688,3 +705,105 @@ impl From<DualAxisData> for Vec2 {
         data.xy
     }
 }
+
+/// The shape of the deadzone for a [`DualAxis`] input.
+///
+/// Uses the x and y thresholds on [`DualAxis`] to set the shape diamensions
+#[derive(Debug, Clone, Copy, Serialize, Deserialize, PartialEq)]
+pub enum DeadZoneShape {
+    /// Deadzone with a cross
+    ///
+    /// Uses the x threshold as the horizontal width and y threshold as the vertical width.
+    Cross {
+        /// The width of the horizontal rectangle.
+        horizontal_width: f32,
+        /// The width of the vertical rectangle.
+        vertical_width: f32,
+    },
+    /// Deadzone with a rectangle.
+    ///
+    /// Uses the x threshold as the width and y threshold as the length.
+    Rect {
+        /// The width of the rectangle.
+        width: f32,
+        /// The height of the rectangle.
+        height: f32,
+    },
+    /// Deadzone with a circle.
+    ///
+    /// Uses the larger of the x or y threshold as the radius
+    Ellipse {
+        /// The horizontal radius
+        radius_x: f32,
+        /// The vertical radius
+        radius_y: f32,
+    },
+}
+
+impl Eq for DeadZoneShape {}
+impl std::hash::Hash for DeadZoneShape {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {}
+}
+
+impl DeadZoneShape {
```

Comment:
@alice-i-cecile 
Is this how you imagined the `impl` to look. Or should `&InputSteams` not be a parameter? Processing the `DualAxis` x and y in `extract_dual_axis_data()` is only really feasible if `Cross` becomes a deadzone instead of a "snapzone", at least from my understanding.

Reply:
Actually, if `Cross` becomes an actual deadzone, then `dual_axis: &DualAxis` and  `input_stream: &InputStreams` both won't be needed to be passed in, and the function can become something like, `point_outside_shape(self, x, y) ` and return a `bool` instead of `DualAxisData`. Then `extract_dual_axis_data()` can get the x, y input values again like:
```rust
fn extract_dual_axis_data(&self, dual_axis: &DualAxis) -> DualAxisData {
    let x = input_stream.input_value(&UserInput::Single(InputKind::SingleAxis(dual_axis.x)), false);
    let y = input_stream.input_value(&UserInput::Single(InputKind::SingleAxis(dual_axis.y)), false);

    if dual_axis.deadzone.point_inside_shape(x, y){
        DualAxisData::new(x, y)
    } else {
        DualAxisData::new(0, 0)
    }
}
```
Which is way cleaner imo. This does have the "loss of functionally" of per-controller snapping, but Bevy kinda does this with livezone in `AxisSettings` in the `GamepadSettings` resource. Plus, if per-controller snapping, instead of global, was needed, it would be quite easy, I think, to add snapping to `DualAxis`.